src/borg/archiver/debug_cmd.py

```diff
@@ -265,6 +266,61 @@ def do_debug_id_hash(self, args, repository, manifest):
         print(id.hex())
         return EXIT_SUCCESS
 
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_parse_obj(self, args, repository, manifest):
+        """parse object file into meta dict and data (decrypting, decompressing)"""
+
+        # get the object from id
+        hex_id = args.id
+        try:
+            id = unhexlify(hex_id)
+            if len(id) != 32:  # 256bit
+                raise ValueError("id must be 256bits or 64 hex digits")
+        except ValueError as err:
+            print("object id %s is invalid [%s]." % (hex_id, str(err)))
+            return EXIT_ERROR
+
```

Comment:
Where can I see the CompressionSpec definition? Is it in `/home/sm/Public/Sock/borg/src/borg/compress.pyx` :sweat_smile:

Also I looked into other validators. Perhaps we can have a `hex_id_validator` like `text_validator` in `/borg/src/borg/helpers/parseformat.py`  if this check is only going to be used in `debug_cmd`s

Reply:
You should have a global search function in your developer toolbox.

If you didn't try it yet, maybe have a look at PyCharm.

The code missing should be give to the argparse `type=...` and it will parse and validate the argument early, before it even invokes the function that implements the command.